components/browser/state/src/main/java/mozilla/components/browser/state/state/BrowserState.kt

```diff
@@ -12,10 +12,13 @@ import mozilla.components.lib.state.State
  * @property tabs the list of open tabs, defaults to an empty list.
  * @property selectedTabId the ID of the currently selected (active) tab.
  * @property customTabs the list of custom tabs, defaults to an empty list.
+ * @property extensions the list of all installed web extensions, defaults to an empty list.
+ * The extensions here represent the default values for all [BrowserState.extensions].
+ * and can be overridden per [SessionState.extensions].
  */
 data class BrowserState(
     val tabs: List<TabSessionState> = emptyList(),
     val selectedTabId: String? = null,
-
-    val customTabs: List<CustomTabSessionState> = emptyList()
+    val customTabs: List<CustomTabSessionState> = emptyList(),
+    val extensions: List<WebExtensionState> = emptyList()
```

Comment:
It seems like we are always traversing this list to find a specific extension. So I was wondering if a `Map` from id to state may make more sense here? 🤔

Reply:
Yes, agree it makes senses 👍

Reply:
Fixed!